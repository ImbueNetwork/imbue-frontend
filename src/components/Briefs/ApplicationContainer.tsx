/* eslint-disable react-hooks/exhaustive-deps */
import Image from 'next/image';
import React from 'react';

// import { FaPaperclip } from 'react-icons/fa';
import { ApplicationContainerProps, applicationStatusId } from '@/model';

export const ApplicationContainer = ({
  application,
  redirectToApplication,
  handleMessageBoxClick,
}: ApplicationContainerProps) => {
  return (
    <div className='flex flex-col px-5 py-5 lg:px-[60px] lg:py-[35px] border-b border-b-imbue-light-purple last:border-b-0 hover:bg-imbue-light-purple-hover cursor-pointer'>
      <div className='flex flex-col gap-5 lg:flex-row lg:items-center justify-between mb-4'>
        <div className='flex items-start gap-5 ml-4 lg:ml-0'>
          <Image
            src={application?.freelancer?.profile_image ?? require('@/assets/images/profile-image.png')}
            className='h-[60px] w-[60px] object-cover rounded-full'
            alt={'profile-picture'}
            height={70}
            width={70}
          />

          <div className='user-id text-content max-w-[70%] lg:max-w-full flex flex-col gap-2'>
            <span className='text-lg'>{application?.freelancer?.display_name}</span>
            <span className='text-sm text-content-primary break-words'>@{application?.freelancer?.username}</span>
            <span className='text-sm w-[fit-content]'>{application?.freelancer?.title}</span>
          </div>
        </div>
        <div className='ctas-container flex flex-row flex-wrap gap-2 pb-4'>
          {/* TODO: Like/unlike feature. On hold */}
          {/* <div className="cta-votes">
                                            <div className="cta-vote">
                                                <FaRegThumbsUp />
                                                Yes
                                            </div>
                                            <div className="cta-vote">
                                                <FaRegThumbsDown />
                                                No
                                            </div>
                                        </div> */}


          <button
            className='border border-imbue-purple rounded-full px-5 text-imbue-purple hover:bg-primary hover:border-primary font-medium py-[6px] md:py-2 !text-sm lg:!text-base'
            onClick={() => redirectToApplication(application?.id)}
          >
            View proposal
          </button>
          <button
            onClick={() =>
              handleMessageBoxClick(
                application?.user_id,
                application?.freelancer
              )
            }
            className='primary-btn in-dark w-button !px-5 !text-sm lg:!text-base'
          >
            Message
          </button>
          <button
            className={`${applicationStatusId[application?.status_id]
              }-btn in-dark text-xs lg:text-base rounded-full py-3 px-3 lg:px-6 lg:py-[10px]`}
          >
            {applicationStatusId[application?.status_id]}
          </button>
        </div>
      </div>

      <div className='flex flex-col gap-[10px] grow'>
        <div className='flex flex-row items-center'>
          {/* <div className="country">
                        <div className="country-flag">
                            <ReactCountryFlag countryCode="us" />
                        </div>
                        <div className="country-name text-grey">
                            United States
                        </div>
                    </div> */}
        </div>

        <p className='text-xl text-content'>
          {application?.name}
        </p>
        <p className='text-base text-content'>
          {
            application?.description?.length > 500
              ? application?.description.substring(0, 500) + "..."
              : application.description
          }
        </p>
        <div className='text-base'>
          {/* <div className='text-imbue-purple-dark'>
            <span className=''>Cover Letter - </span>
            {/* {application.freelancer.bio
                                            .split("\n")
                                            .map((line, index) => (
                                                <span key={index}>{line}</span>
                                            ))} }
            Hello, I would like to help you! I have 4+ years Experience with web
            3, so iâ€™ll make things work properly. Feel free to communicate!
          </div> */}
        </div>

        <div className='flex items-center lg:items-start lg:justify-between'>
          <div className='text-base'>
            {/* <p className='text-base text-imbue-purple-dark'>Attachment(s)</p> */}
            <div className='flex py-3 gap-2'>
              {/* <FaPaperclip color='#b2ff0b' />
              <div className='text-imbue-purple w-4/5 lg:w-full text-xs lg:text-[16px] break-words'>
                https://www.behance.net/abbioty
              </div> */}
            </div>
          </div>
          <div>
            <div className='flex gap-2 flex-col lg:items-center'>
              <span className='text-imbue-purple-dark text-xs lg:text-base'>
                Milestones ({application?.milestones?.length})
              </span>
              <div className='text-imbue-purple text-[16px]'>
                ${Number(application?.required_funds).toLocaleString()}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
