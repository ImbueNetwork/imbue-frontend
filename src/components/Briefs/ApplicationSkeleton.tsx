import { Skeleton } from '@mui/material';
import React from 'react';

const ApplicationSkeleton = () => {
    return (
        <div className='bg-white overflow-hidden rounded-xl'>
            {[1, 2].map((v, i) => (
                <div
                    key={i}
                    className='w-full px-5 py-3 lg:px-10 lg:py-8 border-b last:border-b-0 border-b-imbue-light-purple'
                >
                    <div className='flex justify-between items-center'>
                        <div className='flex w-full items-center gap-4'>
                            <Skeleton
                                className='w-16 h-16'
                                variant='circular'
                                sx={{ fontSize: '1rem' }}
                            />
                            <Skeleton
                                className='w-1/6 h-7'
                                variant='text'
                                sx={{ fontSize: '1rem' }}
                            />
                        </div>
                        <Skeleton
                            className='w-1/6 h-7'
                            variant='text'
                            sx={{ fontSize: '1rem' }}
                        />
                    </div>
                    <div className='flex justify-between'>
                        <Skeleton
                            className='w-5/6'
                            variant='text'
                            sx={{ fontSize: '1rem' }}
                        />
                        <Skeleton
                            className='w-1/12'
                            variant='text'
                            sx={{ fontSize: '1rem' }}
                        />
                    </div>
                    <Skeleton
                        className='w-3/5'
                        variant='text'
                        sx={{ fontSize: '1rem' }}
                    />
                    <Skeleton
                        className='w-1/12'
                        variant='text'
                        sx={{ fontSize: '1rem' }}
                    />
                </div>
            ))}
        </div>
    );
};

export default ApplicationSkeleton;