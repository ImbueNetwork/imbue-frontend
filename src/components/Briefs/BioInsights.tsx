import CancelIcon from '@mui/icons-material/Cancel';
import MarkEmailUnreadOutlinedIcon from '@mui/icons-material/MarkEmailUnreadOutlined';
import VerifiedIcon from '@mui/icons-material/Verified';
import { Alert, Skeleton, Tooltip } from '@mui/material';
import moment from 'moment';
import Image from 'next/image';
import { useRouter } from 'next/router';
import React, { useContext, useEffect, useState } from 'react';
import { FaRegShareSquare } from 'react-icons/fa';

import { checkEnvironment } from '@/utils';

import ChatPopup from '@/components/ChatPopup';

import { copyIcon } from '@/assets/svgs';
import { Brief, Project, User } from '@/model';
import {
  checkIfBriefSaved,
  deleteSavedBrief,
  getBriefApplications,
  saveBriefData,
} from '@/redux/services/briefService';

import { LoginPopupContext, LoginPopupContextType } from '../Layout';
import CountrySelector from '../Profile/CountrySelector';

type BioInsightsProps = {
  redirectToApply: () => void;
  brief: Brief;
  isOwnerOfBrief: boolean | null;
  handleMessageBoxClick: () => void;
  showMessageBox: boolean;
  setShowMessageBox: (_: boolean) => void;
  targetUser: User | null;
  browsingUser: User | null;
  canSubmitProposal: boolean | undefined;
  allClientBriefs: any;
  setSuccess: (_value: boolean) => void;
  setSuccessTitle: (_value: string) => void;
  setError: (_value: any) => void;
  loadingMain: boolean;
};

const BioInsights = ({
  redirectToApply,
  brief,
  browsingUser,
  isOwnerOfBrief,
  handleMessageBoxClick,
  showMessageBox,
  setShowMessageBox,
  targetUser,
  canSubmitProposal,
  allClientBriefs = [],
  setSuccess,
  setError,
  setSuccessTitle,
  loadingMain
}: BioInsightsProps) => {
  const router = useRouter();
  const [copied, setCopied] = useState(false);

  const [loading, setLoading] = useState<boolean>(true);
  const [isSavedBrief, setIsSavedBrief] = useState<boolean>(false);
  // const [clientBriefs, setClientBrief] = useState<Brief[]>([]);
  const [briefApplications, setBriefApplications] = useState<Project[]>([]);
  const lastApplication: Project =
    briefApplications[briefApplications?.length - 1];

  const pendingApplciations: Project[] =
    !briefApplications?.length
      ? []
      : briefApplications?.filter(
        (application) => application?.status_id === 1
      );

  let hint = '';
  if (!canSubmitProposal)
    hint = 'Only verified freelancers can apply to briefs';
  else if (isOwnerOfBrief)
    hint = 'You are not allowed to submit proposal to your own brief';

  const { setShowLoginPopup } = useContext(
    LoginPopupContext
  ) as LoginPopupContextType;

  const acceptedBriefs = allClientBriefs?.acceptedBriefs || []
  const briefsUnderReview = allClientBriefs?.briefsUnderReview || []
  const clientBriefs = [...acceptedBriefs, ...briefsUnderReview]

  const briefWithApplications = allClientBriefs?.briefsUnderReview?.length ? allClientBriefs.briefsUnderReview.filter((brief: Brief) => brief.number_of_applications) : 0
  const hiredCount = allClientBriefs?.acceptedBriefs?.length || 0

  const hireRate = Math.round((hiredCount / (briefWithApplications.length + hiredCount)) * 100)

  useEffect(() => {
    const fetchSavedBriefs = async () => {
      if (!brief?.id) return

      try {
        if (browsingUser?.id) {
          const briefIsSaved = await checkIfBriefSaved(
            brief?.id,
            browsingUser?.id
          );
          setIsSavedBrief(briefIsSaved.isSaved);
        }
        setBriefApplications(await getBriefApplications(brief?.id));
      } catch (error) {
        setError({ message: "Failed to get application data. Please try again. " + error })
      } finally {
        setLoading(false)
      }
    }

    fetchSavedBriefs()
  }, [brief?.id, browsingUser?.id, setError])

  // useEffect(() => {
  //   const setUp = async () => {
  //     if (!allClientBriefs.length) return;

  //     const allBriefs = [...allClientBriefs.acceptedBriefs, ...allClientBriefs.briefsUnderReview];
  //     setClientBrief(allBriefs);
  //   };

  //   setUp();
  // }, [allClientBriefs]);

  const copyToClipboard = () => {
    const textToCopy = checkEnvironment().concat(`${router.asPath}`);

    navigator.clipboard.writeText(textToCopy);
    setCopied(true);

    setTimeout(() => {
      setCopied(false);
    }, 3000);
  };

  const saveBrief = async () => {
    if (!browsingUser?.id)
      return setShowLoginPopup({
        open: true,
        redirectURL: `/briefs/${brief.id}`,
      });

    const resp = await saveBriefData({
      ...brief,
      currentUserId: browsingUser?.id,
    });
    if (resp?.brief_id) {
      setSuccessTitle('Brief Saved Successfully');
      setIsSavedBrief(true);
      setSuccess(true);
    } else {
      setError({ message: 'Brief already Saved' });
    }
  };

  const unsaveBrief = async () => {
    if (!browsingUser?.id) return setError({ message: "No user information found." });

    await deleteSavedBrief(brief.id, browsingUser?.id);
    setIsSavedBrief(false);
    setSuccess(true);
    setSuccessTitle('Brief Unsaved Successfully');
  };

  if (loading || loadingMain) return (
    <div className='brief-insights py-6 px-6 lg:py-10 lg:px-10 mt-[2rem] lg:mt-0 w-full md:w-[35%] relative'>
      <Skeleton variant="text" className='text-xl w-52' />
      <div className='flex gap-2 w-full my-3'>
        <Skeleton variant="text" className='text-xl w-28 h-10 rounded-full' />
        <Skeleton variant="text" className='text-xl w-40 h-10 rounded-full' />
      </div>

      <Skeleton variant="text" className='text-base w-52 mb-3' />
      <Skeleton variant="text" className='text-base w-52 mb-3' />
      <Skeleton variant="text" className='text-base w-52 mb-3' />

      <Skeleton variant="text" className='text-base w-52 mt-6' />
      <Skeleton variant="rounded" className='mt-4 w-full' height={180} />

      <Skeleton variant="text" className='text-base w-52 mt-5' />
      <Skeleton variant="text" className='text-base w-52 mt-2' />
      <Skeleton variant="rounded" className='mt-4 w-full' height={30} />
    </div>
  )

  return (
    <div
      className='brief-insights py-6 px-6 lg:py-10 lg:px-10 mt-[2rem] lg:mt-0 w-full md:w-[35%] relative'
    >
      <div>
        <h3 className='text-imbue-purple-dark !font-normal'>
          Activities on this job
        </h3>

        {!brief?.project_id && (
          <div className='flex gap-3 lg:items-center mt-4 flex-wrap'>
            <button
              onClick={() => (isSavedBrief ? unsaveBrief?.() : saveBrief?.())}
              className={` ${isSavedBrief
                ? 'bg-imbue-coral text-white border-imbue-coral'
                : 'bg-transparent text-content border border-content'
                } rounded-3xl h-[2.48rem] text-base font-normal px-5 max-width-1100px:w-full max-width-500px:w-auto `}
            >
              {isSavedBrief ? 'Unsave' : 'Save'}
            </button>

            <Tooltip
              title={hint}
              arrow
              placement='bottom'
              leaveTouchDelay={5}
              followCursor
            >
              <button
                className={`primary-btn 
              in-dark
              !text-[1rem]
              !font-normal
              flex
              items-center
              gap-2
              !m-0
              !px-4
              ${(!canSubmitProposal || isOwnerOfBrief) &&
                  '!bg-gray-300 !text-gray-400 !cursor-not-allowed'
                  }
              `}
                onClick={() =>
                  canSubmitProposal && !isOwnerOfBrief && redirectToApply()
                }
              >
                Submit a Proposal <FaRegShareSquare />
              </button>
            </Tooltip>
          </div>
        )}
      </div>

      <div className='subsection !mb-[0.75rem]'>
        {brief.verified_only && (
          <p className='text-imbue-purple flex items-center mb-4'>
            <VerifiedIcon className='mr-2' fontSize='small' htmlColor='#38e894' />
            Only verified freelancer can apply
          </p>
        )}
        <div className='brief-insights-stat'>
          <div className='flex items-center text-imbue-purple-dark '>
            Applications:
            <Tooltip
              title='An approximate number of applications for this brief'
              followCursor
              leaveTouchDelay={10}
              className='cursor-pointer'
            >
              <span className='bg-indigo-700 ml-2 h-5 w-5 py-1 px-1.5 cursor-pointer text-xs !text-white rounded-full flex justify-center items-center'>
                ?
              </span>
            </Tooltip>
            <span className='primary-text font-normal ml-2 !text-imbue-lemon'>
              {briefApplications?.length > 0 ? "Less than " : ""}
              {briefApplications?.length || 0}
            </span>
          </div>
        </div>
      </div>

      <div className='subsection !mt-0 !mb-[0.75rem]'>
        <div className='brief-insights-stat'>
          <div className='flex items-center text-imbue-purple-dark'>
            Last applied by freelancers:
            <span className='primary-text font-bold ml-2 !text-imbue-lemon'>
              {lastApplication?.created
                ? moment(lastApplication?.created, 'YYYYMMDD').fromNow()
                : 'Never'}
            </span>
          </div>
        </div>
      </div>

      <div className='subsection !mt-0'>
        <div className='brief-insights-stat'>
          <div className='text-imbue-purple-dark'>
            Currently Interviewing:
            <span className='primary-text font-bold ml-2 !text-imbue-lemon'>
              {pendingApplciations?.length || 0}
            </span>
          </div>
        </div>
      </div>

      {/* Fixme: not implemented in figma design */}
      {/* <div className="subsection">
                <div className="brief-insights-stat">
                    <IoMdWallet className="brief-insight-icon" />
                    <h3>
                        Total Spent <span className="primary-text">$250,000</span>
                    </h3>
                </div>
            </div> */}

      <h3 className='mt-[3rem] text-imbue-purple-dark !font-normal'>
        About Client
      </h3>

      <div className=' bg-imbue-light-purple-three p-[1rem] rounded-[0.5rem] mt-3'>
        <div className='mb-5'>
          <div className='brief-insights-stat flex gap-2 justify-start items-center max-width-1800px:flex-wrap '>
            {targetUser?.web3_address ? (
              <>
                <VerifiedIcon fontSize='small' color='secondary' />
                <span className='font-normal text-imbue-purple-dark text-[1rem] mr-3'>
                  Payment method verified
                </span>
              </>
            ) : (
              <>
                <CancelIcon fontSize='small' color='error' />
                <span className='font-normal text-imbue-purple-dark text-[1rem] mr-3'>
                  Payment method not verified
                </span>
              </>
            )}

            {/* <div>
                  {[4, 4, 4, 4].map((star, index) => (
                    <StarIcon
                      key={`${index}-star-icon`}
                      className={`${index <= 4 && 'primary-icon'}`}
                    />
                  ))}
                </div> */}
          </div>
        </div>

        <div className='mb-5'>
          <div className='brief-insights-stat flex flex-col'>
            <div className='flex gap-2 items-center text-imbue-purple-dark !font-normal'>
              <MarkEmailUnreadOutlinedIcon fontSize='small' />
              <div className='text-lg'>
                <span className='mr-2 '>{clientBriefs.length}</span>
                {`Project${clientBriefs?.length > 1 ? 's' : ''} Posted`}
              </div>
            </div>
            <p className='mt-2 text-imbue-purple text-[1rem]'>
              {hireRate > 0 && <span className='mr-1'>{hireRate}% hire rate,</span>}
              {allClientBriefs?.briefsUnderReview?.length || 0} open job
            </p>
          </div>
        </div>

        {targetUser?.country && (
          <div className='subsection pb-0 !my-0'>
            <div className='brief-insights-stat flex flex-col'>
              <CountrySelector
                user={targetUser}
                setUser={() => null}
                isEditMode={false}
              />
            </div>
          </div>
        )}
        <p className='text-imbue-purple !text-sm'>
          Member since {moment(targetUser?.created).format('MMM DD, YYYY')}
        </p>
        {/* {
          targetUser?.country && (
            <div className='subsection pb-0 !my-0'>
              <div className='brief-insights-stat flex flex-col'>
                <div className='flex items-center'>
                  <Image
                    className='h-4 w-6 object-cover'
                    height={16}
                    width={24}
                    src='https://upload.wikimedia.org/wikipedia/en/a/a4/Flag_of_the_United_States.svg'
                    alt='Flag'
                  />
                  <h3 className='ml-2 text-imbue-purple-dark !font-normal'>
                    United States
                  </h3>
                </div>
                <p className='mt-2 text-imbue-purple text-[1rem]'>
                  Member since Feb 19, 2023
                </p>
              </div>
            </div>
          )
        } */}
      </div>

      <div className='mt-auto'>
        <hr className='separator' />
        <div className='flex flex-col gap-4 mt-5'>
          {!isOwnerOfBrief && (
            <div className='w-full flex gap-3 items-center justify-between'>
              <span className='text-xl text-imbue-purple-dark !font-normal'>
                Meet the hiring team
              </span>
              <button
                onClick={() => handleMessageBoxClick()}
                className='primary-btn in-dark w-button'
                style={{ marginTop: 0 }}
              >
                Message
              </button>
            </div>
          )}
          <h3 className='text-imbue-purple-dark !font-normal'>Job Link</h3>
          <div className='flex items-center gap-2'>
            <div className=' min-h-[2.625rem] rounded-[6.18rem] flex py-2 items-center px-[2rem] bg-imbue-light-purple my-2 w-full'>
              <span className='text-imbue-purple text-[1rem] w-full'>
                {checkEnvironment().concat(`${router.asPath}`)}
              </span>
            </div>

            <Image
              src={copyIcon}
              alt='copyIcon'
              className='cursor-pointer'
              onClick={copyToClipboard}
            />
          </div>
        </div>
        {browsingUser && showMessageBox && (
          <ChatPopup
            showMessageBox={showMessageBox}
            setShowMessageBox={setShowMessageBox}
            targetUser={targetUser}
            browsingUser={browsingUser}
            showFreelancerProfile={false}
          />
        )}
      </div>

      <Alert
        className={`fixed top-28 z-10 transform duration-300 transition-all ${copied ? 'right-5' : '-right-full'
          }`}
        severity='success'
      >
        {`Brief link copied to clipboard`}
      </Alert>
    </div>
  );
};

export default BioInsights;
