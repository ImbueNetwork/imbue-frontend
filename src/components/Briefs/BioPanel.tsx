import VerifiedIcon from '@mui/icons-material/Verified';
import TimeAgo from 'javascript-time-ago';
import en from 'javascript-time-ago/locale/en';
import { useRouter } from 'next/router';
import React, { useState } from 'react';
import { FiEdit } from 'react-icons/fi';

import { Brief } from '@/model';

TimeAgo.addLocale(en);

const timeAgo = new TimeAgo('en-US');

type BioPanelData = {
  brief: Brief;
  projectCategories: string[];
  isOwnerOfBrief?: boolean | null;
  targetUser: any;
};

const BioPanel = ({
  brief,
  projectCategories,
  isOwnerOfBrief,
  targetUser,
}: BioPanelData) => {
  const [expandBreifDesc, setExpandBreifDesc] = useState<number>(500);
  const timePosted = timeAgo.format(new Date(brief.created));
  const router = useRouter();
  return (
    <div className='brief-bio py-5 px-10 max-width-750px:!p-5 max-width-750px:!w-full max-width-1100px:p-[1rem] relative'>
      <div className='mb-6'>
        <div className='flex flex-wrap flex-col items-start mb-1'>
          <div className='header'>
            {brief.verified_only && (
              <p className='text-imbue-purple flex items-center  mb-1.5'>
                <VerifiedIcon className='text-base mr-2' />
                only verified freelancer can apply
              </p>
            )}
            <p className='!text-3xl text-imbue-purple-dark !font-normal'>
              {brief.headline}
            </p>
          </div>

          {isOwnerOfBrief && !brief?.project_id && (
            <button
              className='primary-btn 
              in-dark w-[auto] 
              max-width-750px:!px-4 
              flex 
              items-center 
              gap-2
              my-4
              !self-start
              '
              onClick={() => {
                router.replace(`/briefs/${brief?.id}/edit`);
              }}
            >
              Edit Brief
              <FiEdit size={16} />
            </button>
          )}
        </div>
        <span className='text-sm primary-text !text-imbue-lemon'>
          Posted {timePosted} by{' '}
          <span
            onClick={() => router.push(`/profile/${targetUser.username}`)}
            className='hover:underline cursor-pointer'
          >
            {brief.created_by}
          </span>
        </span>
      </div>

      <div className='subsection break-all'>
        <h3 className='text-imbue-purple-dark !font-normal'>
          Project Description
        </h3>
        <p className='mt-4 text-imbue-purple-dark whitespace-pre-wrap !leading-normal'>
          {brief.description.length > expandBreifDesc
            ? brief.description.substring(0, expandBreifDesc) + ' ...'
            : brief.description}
          {brief.description.length > 500 && (
            <span>
              {brief.description.length > expandBreifDesc ? (
                <button
                  onClick={() => setExpandBreifDesc((prev) => prev + 500)}
                  className='ml-3 w-fit text-sm hover:underline text-imbue-lemon'
                >
                  Show more
                </button>
              ) : (
                <button
                  onClick={() => setExpandBreifDesc(500)}
                  className='ml-3 w-fit text-sm hover:underline text-imbue-lemon'
                >
                  Show Less
                </button>
              )}
            </span>
          )}
        </p>
      </div>

      <div className='subsection'>
        <div className='header'>
          <h3 className='text-imbue-purple-dark !font-normal'>
            Project Category
          </h3>
        </div>
        <div className='list-row'>
          {projectCategories?.map((category, index) => (
            <p
              className='rounded-full bg-imbue-light-purple px-4 py-2 text-[0.875rem] text-imbue-purple'
              key={index}
            >
              {category}
            </p>
          ))}
        </div>
      </div>

      <div className='subsection'>
        <div className='header'>
          <h3 className='text-imbue-purple-dark !font-normal'>
            Key Skills And Requirements
          </h3>
        </div>
        <div className='list-row'>
          {brief.skills?.map((skill, index) => (
            <p
              className='rounded-full bg-imbue-light-purple px-4 py-2 text-[0.875rem] text-imbue-purple'
              key={index}
            >
              {skill.name}
            </p>
          ))}
        </div>
      </div>

      <div className='subsection'>
        <div className='header'>
          <h3 className='text-imbue-purple-dark !font-normal !mt-[2rem]'>
            Project Scope
          </h3>
        </div>
        <span className='text-imbue-purple-dark text-[1rem]'>
          {brief.scope_level}
        </span>
      </div>

      <div className='subsection'>
        <div className='header'>
          <h3 className='text-imbue-purple-dark !font-normal'>
            Experience Level Required
          </h3>
        </div>
        <span className='text-imbue-purple-dark text-[1rem]'>
          {brief.experience_level}
        </span>
      </div>

      <div className='subsection'>
        <div className='header'>
          <h3 className='text-imbue-purple-dark !font-normal'>
            Estimated Length
          </h3>
        </div>
        <span className='text-imbue-purple-dark text-[1rem]'>
          {brief.duration}
        </span>
      </div>

      <div className='subsection'>
        <div className='header'>
          <h3 className='text-imbue-purple-dark !font-normal'>Total Budget</h3>
        </div>
        <span className='text-imbue-purple-dark text-[1rem]'>
          ${Number(brief.budget).toLocaleString()}
        </span>
      </div>
    </div>
  );
};

export default BioPanel;
