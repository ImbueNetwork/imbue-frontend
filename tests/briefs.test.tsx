import {
  render,
  fireEvent,
  waitFor,
  screen,
  RenderResult,
  Matcher,
  SelectorMatcherOptions,
  MatcherOptions,
} from "@testing-library/react";
import React from "react";
import { act } from "@testing-library/react";
import Briefs from "../src/pages/briefs/index";
import {
  getAllBriefs,
  callSearchBriefs,
} from "../src/redux/services/briefService";
import { useRouter } from "next/router";
import {
  amountOfBriefsSubmitted,
  briesData,
  intermidiateExpData,
  projectLengthData,
  searchMockResponse,
} from "./__mocks__/briefsData";
import { renderHook } from "@testing-library/react-hooks";

jest.mock("../src/redux/services/briefService");

jest.mock("../src/pages/briefs/index", () => ({
  getAllBriefs: jest.fn(),
  callSearchBriefs: jest.fn(),
}));

// mock useRouter
jest.mock("next/router", () => ({
  useRouter() {
    return {
      query: {},
    };
  },
}));

describe("Briefs component", () => {
  render(<Briefs />);
  // check if all components are rendered
  expect(screen.getByTestId("result")).toBeInTheDocument();
  // Mock the briefsService.getAllBriefs method to return a fixed array of briefs
  // beforeAll(() => {
  //   const mockGetAllBriefs = getAllBriefs as jest.MockedFunction<
  //     typeof getAllBriefs
  //   >;
  //   mockGetAllBriefs.mockResolvedValue(briesData);
  // });
  // let briefsComponent: RenderResult<
  //   typeof import("@testing-library/dom/types/queries"),
  //   HTMLElement,
  //   HTMLElement
  // >;
  // let appContainer: HTMLElement;
  // let appGetAllByText:
  //   | ((
  //       id: Matcher,
  //       options?: SelectorMatcherOptions | undefined
  //     ) => HTMLElement[])
  //   | ((arg0: string) => any[]);
  // let appQueryAllByTestId:
  //   | ((id: Matcher, options?: MatcherOptions | undefined) => HTMLElement[])
  //   | ((arg0: string) => any[]);
  // beforeEach(async () => {
  //   briefsComponent = await act(async () => render(<Briefs />));
  //   const { container, queryAllByTestId, getAllByText } = await act(async () =>
  //     render(<Briefs />)
  //   );
  //   appContainer = container;
  //   appGetAllByText = getAllByText;
  //   appQueryAllByTestId = queryAllByTestId;
  //   expect(getAllBriefs).toHaveBeenCalled();
  //   await waitFor(() =>
  //     expect(appContainer.getElementsByClassName("brief-title")).toHaveLength(3)
  //   );
  // });
  // it("should render brief items when the component mounts.", async () => {
  //   expect(briefsComponent).toBeTruthy();
  // });
  // it("should render briefs list and filter options", async () => {
  //   // Check that filter options are displayed
  //   await waitFor(() =>
  //     expect(appContainer.getElementsByClassName("filter-option")).toBeTruthy()
  //   );
  // });
  // it("should filter briefs by experience level", async () => {
  //   const mockCallSearchBriefs = callSearchBriefs as jest.MockedFunction<
  //     typeof callSearchBriefs
  //   >;
  //   // get intermidiate checkbox
  //   const intermidiateCheckBox = await waitFor(
  //     () => appQueryAllByTestId("0-1")[0]
  //   );
  //   expect(intermidiateCheckBox).toBeTruthy();
  //   if (appContainer) {
  //     fireEvent.click(intermidiateCheckBox);
  //     mockCallSearchBriefs.mockResolvedValue(intermidiateExpData);
  //     // search for intermidiate briefs
  //     await waitFor(() =>
  //       fireEvent.click(appContainer.getElementsByClassName("tab-item")[0])
  //     );
  //     await waitFor(() =>
  //       expect(appContainer.getElementsByClassName("brief-title")).toHaveLength(
  //         1
  //       )
  //     );
  //   }
  // });
  // it("should filter briefs by number of Briefs submitted", async () => {
  //   const mockCallSearchBriefs = callSearchBriefs as jest.MockedFunction<
  //     typeof callSearchBriefs
  //   >;
  //   const amountSubmittedCheckBox = await waitFor(
  //     () => appQueryAllByTestId("1-2")[0]
  //   );
  //   expect(amountSubmittedCheckBox).toBeTruthy();
  //   if (appContainer) {
  //     fireEvent.click(amountSubmittedCheckBox);
  //     mockCallSearchBriefs.mockResolvedValue(amountOfBriefsSubmitted);
  //     await waitFor(() =>
  //       fireEvent.click(appContainer.getElementsByClassName("tab-item")[0])
  //     );
  //     await waitFor(() =>
  //       expect(appContainer.getElementsByClassName("brief-title")).toHaveLength(
  //         2
  //       )
  //     );
  //   }
  // });
  // it("should filter briefs by project length", async () => {
  //   const mockCallSearchBriefs = callSearchBriefs as jest.MockedFunction<
  //     typeof callSearchBriefs
  //   >;
  //   const projectLengthCheckBox = await waitFor(
  //     () => appQueryAllByTestId("2-0")[0]
  //   );
  //   expect(projectLengthCheckBox).toBeTruthy();
  //   if (appContainer) {
  //     fireEvent.click(projectLengthCheckBox);
  //     mockCallSearchBriefs.mockResolvedValue(projectLengthData);
  //     await waitFor(() =>
  //       fireEvent.click(appContainer.getElementsByClassName("tab-item")[0])
  //     );
  //     await waitFor(() =>
  //       expect(appContainer.getElementsByClassName("brief-title")).toHaveLength(
  //         1
  //       )
  //     );
  //   }
  // });
  // test("test brief rendering and matching the snapshot", async () => {
  //   expect(screen.queryAllByText("briefs found")[0]).toMatchSnapshot();
  // });
  // test("onSearch should filter briefs based on search input and selected checkboxes", async () => {
  //   // Mock the elements and values needed for the test
  //   const mockCallSearchBriefs = callSearchBriefs as jest.MockedFunction<
  //     typeof callSearchBriefs
  //   >;
  //   const experienceCheckBox = await waitFor(
  //     () => appQueryAllByTestId("0-1")[0]
  //   );
  //   const submittedCheckBox = await waitFor(
  //     () => appQueryAllByTestId("1-2")[0]
  //   );
  //   const searchInput = appContainer.getElementsByClassName("search-input")[0];
  //   expect(experienceCheckBox).toBeTruthy();
  //   expect(submittedCheckBox).toBeTruthy();
  //   expect(searchInput).toBeTruthy();
  //   if (appContainer) {
  //     // fire checkboxes event
  //     fireEvent.click(experienceCheckBox);
  //     fireEvent.click(submittedCheckBox);
  //     // input search text
  //     fireEvent.change(searchInput, {
  //       target: { value: "briefThree" },
  //     });
  //     mockCallSearchBriefs.mockResolvedValue(searchMockResponse);
  //     // search for input brief
  //     await waitFor(() =>
  //       fireEvent.click(appContainer.getElementsByClassName("tab-item")[0])
  //     );
  //     await waitFor(() =>
  //       expect(appGetAllByText("briefThree")[0]).toBeTruthy()
  //     );
  //   }
  // });
});
