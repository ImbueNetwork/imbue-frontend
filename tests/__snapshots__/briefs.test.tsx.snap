// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Briefs component test brief rendering and matching the snapshot 1`] = `
<span>
   briefs found
</span>
`;
